src/v0/destinations/am/transform.js

```diff
@@ -131,8 +131,10 @@ function createRevenuePayload(message, rawPayload) {
 
 function updateTraitsObject(property, traitsObject, actionKey) {
   const propertyToUpdate = getValueFromMessage(traitsObject, property);
-  traitsObject[actionKey][property] = propertyToUpdate;
-  deleteObjectProperty(traitsObject, property);
+  if (property && property.length > 0) {
+    traitsObject[actionKey][property] = propertyToUpdate;
```

Comment:
Based on the error message, it seems like `traitsObject[actionKey]` was undefined. Let's add that to the validation check.